corehq/apps/case_migrations/views.py

```diff
@@ -81,6 +91,15 @@ def migration_restore(request, domain, case_id):
             # Formplayer will be creating these cases for the first time, so
             # include create blocks
             content.append(get_case_element(case, ('create', 'update'), V2))
+        if ADD_LIMITED_FIXTURES_TO_SMS_WORKFLOW.enabled(domain):
+            _add_limited_fixtures(domain, case_id, content)
         payload = content.get_fileobj()
 
     return RestoreResponse(payload).get_http_response()
+
+
+def _add_limited_fixtures(domain, case_id, content):
+    serializer = FlatLocationSerializer()
+    data_fields = get_location_data_fields(domain)
+    restore_user = MockCaseAsRestoreUser(user_id=case_id)
+    content.extend(serializer.get_xml_nodes('locations', restore_user, SQLLocation.active_objects, data_fields))
```

Comment:
Note: We can't use the usual location fixture providers used in restore, https://github.com/dimagi/commcare-hq/blob/1b5625a828f5ad1772fa721687704f0bfd8a5e9e/corehq/apps/locations/fixtures.py#L234-L239  because that needs a restore user, so assuming the right way would be to just generate the content here. Also seems in line with the way we are generating content for cases above.

Reply:
I think that's right - makes sense for this code to directly control what queryset of locations is going to be synced, with the serialization code shared.